### PR TITLE
Dasherize all api fields

### LINKED ISSUE #3753: Dasherize all fields in the api
- Dasherize all fields as shown in http://marshmallow-jsonapi.readthedocs.io/en/latest/quickstart.html#inflection
- Cleanup unused dependencies.

### PR DESCRIPTION
**Pull request by shubham-padia**
Cleanup unused dependencies

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Dasherize all fields as shown in http://marshmallow-jsonapi.readthedocs.io/en/latest/quickstart.html#inflection
- Cleanup unused dependencies.

Fixes #3753

### CODE REVIEW

**Comment by shubham-padia** [comment]
Do not merge, ensuring 
> Also ensure the URLs are also dasherized as per http://jsonapi.org/recommendations/#naming

**Comment by shubham-padia** [comment]
Dasherized all urls, please review

**Comment by niranjan94** [comment]
@fossasia/open-event-server-devs please ensure all your currently open and future PRs take care of the changes made here.

File: app/api/call_for_papers.py [hunk 1]
```diff
@@ -1,15 +1,14 @@
+import pytz
+import dateutil.parser
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
```
Comment (poush) [inline app/api/call_for_papers.py]: blank line here, separating python modules from third party modules.

Comment (shubham-padia) [inline app/api/call_for_papers.py]: Thanks @poush

**Pull request merged**